examples/calibrateTask.py

```diff
@@ -52,81 +58,97 @@ def loadData(pixelScale=1.0):
+        """
+        wcs = exposure.getWcs()
+        refSchema = LoadReferenceObjectsTask.makeMinimalSchema(
+            filterNameList = [FilterName],
+            addIsPhotometric = True,
+        )
+        refCat = afwTable.SimpleCatalog(refSchema)
+        refFluxKey = refSchema[FilterName + "_flux"].asKey()
+        refIsPhotoKey = refSchema["photometric"].asKey()
+
+        matches = lsst.afw.table.ReferenceMatchVector()
+        for src in sourceCat:
+            flux = 1e-3*src.getPsfFlux()*np.random.normal(1.0, 2e-2)
+            refObj = refCat.addNew()
+            refObj.set(refFluxKey, flux)
+            refObj.setCoord(wcs.pixelToSky(src.getCentroid()))
+            refObj.set(refIsPhotoKey, True)
+            match = lsst.afw.table.ReferenceMatch(refObj, src, 0)
+            matches.append(match)
 
         return pipeBase.Struct(
-            matches=matches,
-            matchMeta=md
-            )
+            refCat = refCat,
+            matches = matches,
+            matchMeta = createMatchMetadata(exposure),
+        )
 
 def run(display=False):
     #
-    # Create the task
+    # Create the tasks
     #
+    charImageConfig = CharacterizeImageTask.ConfigClass()
+    charImageTask = CharacterizeImageTask(config=charImageConfig)
+
     config = CalibrateTask.ConfigClass()
-    config.initialPsf.pixelScale = 0.185 # arcsec per pixel
-    config.initialPsf.fwhm = 1.0
     config.astrometry.retarget(MyAstrometryTask)
     calibrateTask = CalibrateTask(config=config)
-    #
-    # Process the data
-    #
-    exposure = loadData(config.initialPsf.pixelScale)
-    result = calibrateTask.run(exposure)
 
-    exposure0, exposure = exposure, result.exposure
-    sources = result.sources
+    # load the data
+    # Exposure ID and the number of bits required for exposure IDs are usually obtained from a data repo,
+    # but here we pick reasonable values (there are 64 bits to share between exposure IDs and source IDs).
+    exposure = loadData()
+    exposureIdInfo = ExposureIdInfo(expId=1, expBits=5)
 
-    if display:                         # display on ds9 (see also --debug argparse option)
-        frame = 1
-        ds9.mtv(exposure, frame=frame)
+    # characterize the exposure to repair cosmic rays and fit a PSF model
+    # display now because CalibrateTask modifies the exposure in place
+    charRes = charImageTask.characterize(exposure=exposure, exposureIdInfo=exposureIdInfo)
+    if display:
+        displayFunc(charRes.exposure, charRes.sourceCat, frame=1)
 
-        with ds9.Buffering():
-            for s in sources:
-                xy = s.getCentroid()
-                ds9.dot('+', *xy, ctype=ds9.CYAN if s.get("flags_negative") else ds9.GREEN, frame=frame)
+    # calibrate the exposure
+    calRes = calibrateTask.calibrate(exposure=charRes.exposure, exposureIdInfo=exposureIdInfo)
+    if display:
+        displayFunc(calRes.exposure, calRes.sourceCat, frame=2)
+
+def displayFunc(exposure, sourceCat, frame):
```

Comment:
Why did you need to pull this out into its own function?  Please don't make unnecessary changes.